MyMesh Refactor and Advert Button

hank opened the pull request:
This branch does a few things:

* Pulls the companion MyMesh class into a header and CPP file and makes the right externed globals for the Button, MyMesh, main and UITask files to actually integrate with each other
* Adds a double press action to button behavior on all devices that sends an advert
* Fixes all the companion ini files to include all files in the companion directory so there shouldn't be linker errors
* Ports a self advert func from repeater since that was missing in the MyMesh for companion

FIRMWARE_VERSION ends up in MyMesh.h, which can be moved if desired. main.cpp just isn't a great place for it. This brings up a potential future task of cleaning up the MyMesh classes in the different example codebases to make a proper class structure that would support Repeater, Room Server and Companion (and other) implementations.

I just tested the doublepress advert on a tbeam sx1276 and it works great, shows up immediately on another node's discover list.

```
DEBUG: UITask: any press triggered
DEBUG: UITask: any press triggered
DEBUG: UITask: double press triggered, sending advert
DEBUG: Advert sent!
```

446564 commented:
I'll test this on the g2 nano tonight

ripplebiz commented on examples/companion_radio/UITask.h:
```diff
@@ -1,4 +1,5 @@
-#pragma once
+#ifndef UI_TASK_H
+#define UI_TASK_H
```
'#pragma once'  does this, and is very widely supported now.

ripplebiz commented on variants/t114/platformio.ini:
```diff
@@ -75,7 +75,7 @@ build_flags =
 ;  -D ENABLE_PRIVATE_KEY_EXPORT=1
 ;  -D MESH_PACKET_LOGGING=1
-;  -D MESH_DEBUG=1
+  -D MESH_DEBUG=1
```
try not to leave these uncommented/enabled when committing

ripplebiz commented on examples/companion_radio/MyMesh.h:
```diff
@@ -0,0 +1,283 @@
+
+#define  PUBLIC_GROUP_PSK  "izOH6cXN6mrJ5e26oRXNcg=="
+
+#define CMD_APP_START              1
```
the CMD_ and RESP_ codes don't need to be known by other modules, so can be moved to the .cpp file

hank commented:
* pragma onced
* Moved CMD/RESP macros as requested
* Added clang-format to the repo which vscode will use to format code, and set it up to match what I saw in the repo with the braces and spacing. Tweak to your heart's content!
* Reformatted all my changes
* Removed debug macro from ini
* Merged in current dev, which required porting a change from main.cpp into MyMesh.cpp because of refactor.

Please review! Thanks!

hank commented:
* Reverted changes to Button files as requested
* Reverted changes to simple_secure_chat
* Switched AllowShortIfStatementsOnASingleLine to true to clang format rules
* Switched BreakBeforeBraces to Attach to make all opening braces occur on the same line
* Reformatted MyMesh.cpp as an example of clang-format changes. Can update more if needed.
* Did my best to minimize changes to the repo
* Built many companion firmwares and tested one to ensure things aren't completely broken, more testing recommended.
